android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java

```diff
@@ -0,0 +1,226 @@
+            File dest = new File(exportDir, source.getName());
+            int n = 0;
+            String fullSourceName = source.getName();
+            String sourceName, sourceExt;
+            if (fullSourceName.indexOf('.') >= 0) {
+                sourceName = fullSourceName.substring(0, fullSourceName.lastIndexOf('.'));
+                sourceExt = fullSourceName.substring(fullSourceName.lastIndexOf('.'));
+            } else {
+                sourceName = fullSourceName;
+                sourceExt = "";
+            }
+            while (!dest.createNewFile()) {
+                dest = new File(exportDir, sourceName + "_" + (n++) + sourceExt);
+            }
+            input = new FileInputStream(source).getChannel();
+            output = new FileOutputStream(dest).getChannel();
+            output.transferFrom(input, 0, input.size());
+            input.close();
+            output.close();
+
+            MediaScannerConnection.scanFile(
+                    mContext,
+                    new String[]{dest.getAbsolutePath()},
+                    null,
+                    new MediaScannerConnection.OnScanCompletedListener() {
+                        @Override
+                        public void onScanCompleted(String path, Uri uri) {
+                            if (uri != null) {
+                                mPromise.resolve(uri.toString());
+                            } else {
+                                mPromise.reject(ERROR_UNABLE_TO_SAVE, "Could not add image to gallery");
+                            }
+                        }
+                    });
+        } catch (IOException e) {
+            mPromise.reject(e);
+        } finally {
+            if (input != null && input.isOpen()) {
+                try {
+                    input.close();
+                } catch (IOException e) {
+                    FLog.e(ReactConstants.TAG, "Could not close input channel", e);
+                }
+            }
+            if (output != null && output.isOpen()) {
+                try {
+                    output.close();
+                } catch (IOException e) {
+                    FLog.e(ReactConstants.TAG, "Could not close output channel", e);
+                }
+            }
+        }
+    }
+
+    private void manageAndroidQ(File source) throws IOException {
+        FileChannel input = new FileInputStream(source).getChannel();
+        String fullSourceName = source.getName();
+
+        ContentResolver resolver = mContext.getContentResolver();
+        Uri videoCollection = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;;
+
+        ContentValues contentDetails = new ContentValues();
+        contentDetails.put(MediaStore.Video.Media.DISPLAY_NAME, fullSourceName);
+        contentDetails.put(MediaStore.Video.Media.IS_PENDING, 1);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            String relativePath = Environment.DIRECTORY_MOVIES + File.separator + mContext.getPackageName();
```

Comment:
Shouldn't `mContext.getPackageName()` be album name provided.

Reply:
for what I read it has to be the package name of the application and of course the album name should be added to it. that's something missing.